app/models/foreman_fog_proxmox/proxmox_images.rb

```diff
@@ -48,11 +48,22 @@ def templates
       volumes = []
       nodes.each do |node|
         storages(node.node).each do |storage|
-          # fetches volumes of QEMU servers for images
-          volumes += storage.volumes.list_by_content_type('images')
+          # Skip disabled storages (enabled == 0 or nil)
+          unless storage.enabled.to_i == 1
+            logger.warn("Skipping disabled storage #{storage.identity} on #{node.node}")
+            next
+          end
 
-          # fetches volumes of KVM containers for images
+          # Fetch QEMU and LXC template images
+          volumes += storage.volumes.list_by_content_type('images')
           volumes += storage.volumes.list_by_content_type('rootdir')
+        rescue Excon::Error::InternalServerError => e
```

Comment:
why is this necessary if there is also the block in https://github.com/theforeman/foreman_fog_proxmox/pull/433/files#diff-27f981f2a27945b7d516f049a4b2057837ba64ee9c6025792fab958e2670253aR52?

Reply:
to catch error if storage fetch fails due to permission issue or if the compute resource is not able to fetch storages from other nodes.